src/colvar_UIestimator.h

```diff
@@ -687,45 +688,53 @@ namespace UIestimator {
                 std::string count_filename = filename[i] + ".UI.count";
                 std::string grad_filename = filename[i] + ".UI.grad";
 
-                std::ifstream count_file(count_filename.c_str(), std::ios::in);
-                std::ifstream grad_file(grad_filename.c_str(), std::ios::in);
+                std::istream *count_file_p =
+                    proxy->input_stream(count_filename, "count filename");
+                std::istream *grad_file_p =
+                    proxy->input_stream(grad_filename, "gradient filename");
 
-                count_file >> sharp >> dimension_temp;
-                grad_file >> sharp >> dimension_temp;
+                if (count_file_p != NULL && grad_file_p != NULL) {
```

Comment:
Could this test be reversed and the function exit upon failure, as done in other instances? This would avoid wrapping most of the function in the test body, reducing nesting depth and improving readability.

Reply:
Sure, why not?